omnibus/config/software/datadog-agent-integrations.rb

```diff
@@ -83,47 +80,73 @@
+    " --hash=sha256:9515fe0a94e823fd90b08d22de45d7bde57c90edce705b22f5e1ecf7e1b653c8"
+    
+    all_reqs_file.close
+
     # Install all the requirements
+    # Install all the build requirements
+     if windows?
+       pip_args = "install --require-hashes -r #{project_dir}/check_requirements.txt"
+       command "#{windows_safe_path(install_dir)}\\embedded\\scripts\\pip.exe #{pip_args}"
+     else
+       build_env = {
+         "LD_RUN_PATH" => "#{install_dir}/embedded/lib",
+         "PATH" => "#{install_dir}/embedded/bin:#{ENV['PATH']}",
+       }
+       pip "install --require-hashes -r #{project_dir}/check_requirements.txt", :env => build_env
+     end
+
+    # Set frozen requirements without pip-tools 
+    pip "freeze > #{install_dir}/constraint_requirements.txt"
+
+    # Install all the build requirements
     if windows?
-      pip_args = "install  -r #{project_dir}/check_requirements.txt"
+      pip_args = "install pip-tools==#{PIPTOOLS_VERSION}"
       command "#{windows_safe_path(install_dir)}\\embedded\\scripts\\pip.exe #{pip_args}"
     else
       build_env = {
         "LD_RUN_PATH" => "#{install_dir}/embedded/lib",
         "PATH" => "#{install_dir}/embedded/bin:#{ENV['PATH']}",
       }
-      pip "install -r #{project_dir}/check_requirements.txt", :env => build_env
+      pip "install pip-tools==#{PIPTOOLS_VERSION}", :env => build_env
     end
 
-    # Set frozen requirements
-    pip "freeze > #{install_dir}/agent_requirements.txt"
-
     # Windows pip workaround to support globs
     python_bin = "\"#{windows_safe_path(install_dir)}\\embedded\\python.exe\""
-    python_pip = "pip install -c #{windows_safe_path(install_dir)}\\agent_requirements.txt #{windows_safe_path(project_dir)}"
-    python_pip_req = "pip install -c #{windows_safe_path(install_dir)}\\agent_requirements.txt -r #{windows_safe_path(project_dir)}"
+    python_pip_no_deps = "pip install -c #{windows_safe_path(install_dir)}\\constraint_requirements.txt --no-deps #{windows_safe_path(project_dir)}"
+    python_pip_req = "pip install -c #{windows_safe_path(install_dir)}\\constraint_requirements.txt --require-hashes -r #{windows_safe_path(project_dir)}"
+    python_pip_uninstall = "pip uninstall -y"
 
-    pinned_requirements = "#{project_dir}/datadog_checks_base/requirements.in"
     if windows?
-      command("#{python_bin} -m #{python_pip}\\datadog_checks_base")
-      if File.exist? pinned_requirements
-        command("#{python_bin} -m #{python_pip_req}\\datadog_checks_base\\requirements.in")
+      command("#{python_bin} -m #{python_pip_no_deps}\\datadog_checks_base")
+      command("#{python_bin} -m piptools compile --generate-hashes --output-file #{windows_safe_path(project_dir)}\\static_requirements.txt #{windows_safe_path(project_dir)}\\datadog_checks_base\\datadog_checks\\data\\agent_requirements.in")
+
+      # Uninstall the deps that pip-compile installs so we don't include them in the final artifact
+      for dep in UNINSTALL_PIPTOOLS_DEPS
```

Comment:
Let's do this the way it's done in A5, where we check `UNINSTALL_PIPTOOLS_DEPS` were not part of the agent constraints. It's just safer, we never know what other software definitions might do.

Reply:
There are some weird issues with Ruby in terms of doing a File.read on the constraints file to make sure those dependencies aren't there.